hatch_build.py

```diff
@@ -0,0 +1,31 @@
+import logging
+import os
+import sys
+
+from hatch_jupyter_builder.plugin import JupyterBuildHook
+
+LOGGER = logging.getLogger(__name__)
+
+
+class CustomBuildHook(JupyterBuildHook):
+    """
+    We use a custom build hook to detect pre-commit environments. In those
+    environments there is no need to build the extension (and, in addition
+    the build will fail if npm is not available, which is often the case
+    of pre-commit environments).
+    """
+
+    def initialize(self, version, _):
+        if "/.cache/pre-commit/".replace("/", os.path.sep) in sys.executable:
```

Comment:
This is a good idea. But I am not quite sure how are you detecting if the current installation is done by `pre-commit` and not a regular installation.

Reply:
Well, so far I have not found a better way than checking whether the Python path contains `/.cache/pre-commit/` (I still need to check how it looks like on Windows)